File: app/src/main/java/com/all/dwcarpentry/ui/recyclerviews/HousesRecyclerViewAdapter.java
//package com.all.dwcarpentry.ui.recyclerviews;
//
//import android.util.Log;
//import android.view.LayoutInflater;
//import android.view.ViewGroup;
//
//import androidx.annotation.NonNull;
//import androidx.fragment.app.Fragment;
//
//import com.all.dwcarpentry.R;
//import com.all.dwcarpentry.data.House;
//import com.firebase.ui.database.paging.DatabasePagingOptions;
//import com.firebase.ui.database.paging.FirebaseRecyclerPagingAdapter;
//import com.firebase.ui.database.paging.LoadingState;
//
//import java.util.ArrayList;
//import java.util.HashMap;
//
//public class HousesRecyclerViewAdapter extends FirebaseRecyclerPagingAdapter<House, HouseViewHolder>
//{
//    private Fragment parentFragment;
//    private OnHouseCardClickedListener onHouseCardClickedListener;
//    public HousesRecyclerViewAdapter(@NonNull DatabasePagingOptions<House> options, Fragment parentFragment, OnHouseCardClickedListener listener)
//    {
//        super(options);
//        this.parentFragment = parentFragment;
//    }
//
//    @Override
//    protected void onBindViewHolder(@NonNull HouseViewHolder holder, int position, @NonNull House model)
//    {
//        holder.bind(model, parentFragment);
//    }
//
//    @Override
//    protected void onLoadingStateChanged(@NonNull LoadingState state)
//    {
//
//    }
//
//    @NonNull
//    @Override
//    public HouseViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
//    {
//        Log.v("TEST", "OnCreateViewHolder was called.");
//        return new HouseViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.house_card, parent, false), onHouseCardClickedListener);
//    }
//
//    public interface OnHouseCardClickedListener
//    {
//        void onHouseCardClicked(String houseKey);
//    }
//}

File: app/src/main/java/com/all/dwcarpentry/ui/recyclerviews/HousesRecyclerViewAdapter.kt
package com.all.dwcarpentry.ui.recyclerviews

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import com.all.dwcarpentry.R
import com.all.dwcarpentry.data.room.House

class HousesRecyclerViewAdapter(private val housesData: MutableList<House>, private val fragment: Fragment, private val onHouseCardClickedListener : OnHouseCardClickedListener) : RecyclerView.Adapter<HouseViewHolder>()
{
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HouseViewHolder
    {
        return HouseViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.house_card, parent, false),
            onHouseCardClickedListener, parent.context.contentResolver)
    }
    override fun getItemCount(): Int
    {
        return housesData.size
    }
    override fun onBindViewHolder(holder: HouseViewHolder, position: Int)
    {
        holder.bind(housesData[position])
    }
    fun updateHouses(newHouses: List<House>)
    {
        housesData.clear()
        housesData.addAll(newHouses)
        notifyDataSetChanged()
    }
}

interface OnHouseCardClickedListener
{
    fun onHouseCardClicked(houseId: Int);
}